cassiopeia/data/CassiopeiaTree.py

```diff
@@ -517,6 +513,88 @@ def children(self, node: str) -> List[str]:
         self.__check_network_initialized()
         return [v for v in self.__network.successors(node)]
 
+    def __remove_node(self, node) -> None:
+        """Private method to remove node from tree.
+        
+        Args:
+            node: A node in the tree to be removed
+
+        Raises:
+            CassiopeiaTreeError if the tree is not initialized.
+        """
+        self.__check_network_initialized()
+
+        self.__network.remove_node(node)
+    
+    def __add_node(self, node) -> None:
+        """Private method to add node to tree.
+        
+        Args:
+            node: A node to be added to the tree.
+            
+        Raises:
+            CassiopeiaTreeError if the tree is not initialized.
+        """
+        self.__check_network_initialized()
+        
+        self.__network.add_node(node)
+
+    def __remove_edge(self, u, v) -> None:
+        """Private method to remove edge from tree.
+        
+        Args:
+            u: The source node of the directed edge to be removed
+            v: The sink node of the directed edge to be removed
+            
+        Raises:
+            CassiopeiaTreeError if the tree is not initialized.
+        """
+        self.__check_network_initialized()
+
+        self.__network.remove_edge(u, v)
+
+    def __add_edge(self, u, v) -> None:
+        """Private method to add edge to tree.
+
+        Args:
+            u: The source node of the directed edge to be added
+            v: The sink node of the directed edge to be added
+            
+        Raises:
+            CassiopeiaTreeError if the tree is not initialized.
+        """
+        self.__check_network_initialized()
+
+        self.__network.add_edge(u, v)
+
+    def out_degree(self, node) -> None:
+        """Gets the out degree of a node in the tree.
+        
+        Args:
+            node: A node in the tree.
+            
+        Raises:
+            CassiopeiaTreeError if the tree is not initialized.
+        """
+
+        self.__check_network_initialized()
+
+        return self.__network.out_degree(node)
+
+    def in_degree(self, node) -> None:
+        """Gets the in degree of a node in the tree.
+
+        Args:
+            node: A node in the tree.
+            
+        Raises:
+            CassiopeiaTreeError if the tree is not initialized.
+        """
+
+        self.__check_network_initialized()
+
+        return self.__network.in_degree(node)
+
```

Comment:
Why do we need these utilities? It looks like you're using them to check for roots and leaves, but we already have functions to do this -- `is_root`, `is_leaf`, and `is_internal_node`. If there's not another purpose for them, I propose we remove them.

(Also - if you're curious about the out degree, you can get this by len(self.children(n))`. By definition, all nodes should have _at most_ 1 in degree (the exception being the root))